src/main/cs/Infinispan/HotRod/Config/ConfigurationBuilder.cs

```diff
@@ -123,6 +124,12 @@ public ClusterConfigurationBuilder AddCluster(string clusterName)
             return new ClusterConfigurationBuilder(this, builder.AddCluster(clusterName));
         }
 
+        public ConfigurationBuilder FailOverBalancingStrategy(Infinispan.HotRod.Config.FailOverRequestBalancingStrategyProducerDelegate d)
```

Comment:
We could probably call it FailOverBalancingStrategyProducer because it really accepts a producer, not the strategy itself

Reply:
I wonder why we need to pass a producer rather than the instance itself but this is probably related to how the C++ code is implemented, right?

Reply:
Right, I'm changing the name. My initial idea was to use the same name as in c++.

Yes C++ wants a producer.

Reply:
We could use the same name as in C++: BalancingStrategyProducer